scripts/sentry-node/docker-compose.yml

```diff
@@ -0,0 +1,138 @@
+# Docker compose file to simulate a sentry node setup.
+#
+#
+# Setup:
+#
+# Validator A is not supposed to be connected to the public internet. Instead it
+# connects to a sentry node (sentry-a) which connects to the public internet.
+# Validator B can reach validator A via sentry node A and vice versa.
+#
+#
+# Usage:
+#
+# 1. Build `target/debug/substrate` binary: `cargo build`
+#
+# 2. Start networks and containers: `sudo docker-compose -f scripts/sentry-node/docker-compose.yml up`
+#
+# 3. Reach:
+#   - polkadot/apps on localhost:3000
+#   - validator-a: localhost:9944
+#   - validator-b: localhost:9945
+#   - sentry-a: localhost:9946
+
+version: "3.7"
+services:
+
+  validator-a:
+    ports:
+      - "9944:9944"
+    volumes:
+      - ../../target/debug/substrate:/usr/local/bin/substrate
+    image: parity/substrate
+    networks:
+      - network-a
+    command:
+      # Local node id: QmRpheLN4JWdAnY7HGJfWFNbfkQCb6tFf4vvA6hgjMZKrR
+      - "--node-key"
+      - "0000000000000000000000000000000000000000000000000000000000000001"
+      - "--base-path"
+      - "/tmp/alice"
+      - "--chain=local"
+      - "--key"
+      - "//Alice"
+      - "--port"
+      - "30333"
+      - "--validator"
+      - "--name"
+      - "AlicesNode"
+      - "--bootnodes"
+      - "/dns4/validator-b/tcp/30333/p2p/QmSVnNf9HwVMT1Y4cK1P6aoJcEZjmoTXpjKBmAABLMnZEk"
+      # Not only bind to localhost.
+      - "--ws-external"
+      - "--rpc-external"
+      # - "--log"
+      # - "sub-libp2p=trace"
+      # - "--log"
+      # - "afg=trace"
+      - "--no-telemetry"
+      - "--rpc-cors"
+      - "all"
+
+  sentry-a:
+    image: parity/substrate
+    ports:
+      - "9946:9944"
+    volumes:
+      - ../../target/debug/substrate:/usr/local/bin/substrate
+    networks:
+      - network-a
+      - internet
+    command:
+      # Local node id: QmV7EhW6J6KgmNdr558RH1mPx2xGGznW7At4BhXzntRFsi
+      - "--node-key"
+      - "0000000000000000000000000000000000000000000000000000000000000003"
+      - "--base-path"
+      - "/tmp/sentry"
+      - "--chain=local"
+      # Don't configure a key, as sentry-a is not a validator.
+      # - "--key"
+      # - "//Charlie"
+      - "--port"
+      - "30333"
+      # sentry-a is not a validator.
+      # - "--validator"
+      - "--name"
+      - "CharliesNode"
+      - "--bootnodes"
```

Comment:
Maybe use `--reserved-nodes` here instead? Same for the other nodes configs.

Reply:
Would you mind adding some more details @andresilva?

If I understand correctly, validator addresses can change at runtime. Thus if the address of *validator-b* changes, *sentry-a* would need to be restarted. I would suggest only interconnecting *validator-a* and *sentry-a* as reserved nodes and *sentry-a* and *validator-b* as bootstrap nodes, given that *a* and *b* are controlled by different entities.

What do you think?

On a side note, I am currently looking into how a user can tell a validator node to keep *x* amount of connections open to other validator nodes. As the validator address set can change at runtime, these long-lived connections would also be adjusted along the way.

Reply:
In this config `A` connects to `B` (through bootnodes). The sentry connects to `A` and `B`, and `B` also tries to connect with `A`.

My understanding is that in a real scenario the validator nodes that are using a sentry will only be connected to the sentry (or multiple sentries). So I think in this config `Validator-A` should set `Sentry-A` as a reserved node (so that it only connects to this node), and the rest of the nodes (`Sentry-A` and `Validator-B`) can add everyone as bootnode (even though `Validator-B` won't be able to connect to `Validator-A`).